core/src/test/java/mediumtest/http/ProxyMediumTests.java

```diff
@@ -73,19 +73,19 @@ class ProxyMediumTests {
   @BeforeEach
   void configureProxy(TestInfo info) {
     if (info.getTags().contains(PROXY_AUTH_ENABLED)) {
-      proxyMock.stubFor(get(urlMatching(".*"))
+      proxyMock.stubFor(get(urlMatching("/api/rules/.*"))
```

Comment:
Unrelated change?

Reply:
By allowing everything, it sends a request to `/api/system/status` due to the changes I made. This means the tests fail when verifying if `/api/rules/show` got requested.
The easy and quick fix was to allow only `/api/rules` because we do not need anything else in this test

Reply:
Following the main comment that I didn't see, caching could be done but then, in case I switch from `Connection_A` -> `Connection_B` -> `Connection_A`, I would expect to see the message again for `Connection_A`?